plugins/validate/__init__.py

```diff
@@ -0,0 +1,126 @@
+        validatorlib = "ansible_collections.{corg}.{cname}.plugins.validate.{plugin}".format(
+            **cref
+        )
+
+        validatordoc = getattr(import_module(validatorlib), "DOCUMENTATION")
+        if validatordoc:
+            self._set_sub_plugin_options(validatordoc)
+
+    def _set_sub_plugin_options(self, doc):
+        params = {}
+        argspec = deepcopy(doc)
+        argspec_obj = yaml.load(argspec, SafeLoader)
+        options = argspec_obj.get("options", {})
+
+        if not options:
+            return None
+
+        for option_name, option_value in iteritems(options):
+
+            option_var_name_list = option_value.get("vars", [])
+            option_env_name_list = option_value.get("env", [])
+
+            # check if plugin configuration option passed as kwargs
+            # valid for lookup, filter, test plugins or pass through
+            # variables if supported by the module.
+            if option_name in self._kwargs:
+                params[option_name] = self._kwargs[option_name]
+                continue
+
+            # check if plugin configuration option passed in task vars eg.
+            #  vars:
+            #  - name: ansible_validate_jsonschema_draft
+            #  - name: ansible_validate_jsonschema_draft_type
+            if option_var_name_list and (option_name not in params):
+                for var_name_entry in to_list(option_var_name_list):
+                    if not isinstance(var_name_entry, dict):
+                        raise AnsibleError(
+                            "invalid type '%s' for the value of '%s' option,"
+                            " should to be type dict"
+                            % (type(var_name_entry), var_name_entry)
+                        )
+                    var_name = var_name_entry.get("name")
+                    if var_name and var_name in self._plugin_vars:
+                        params[option_name] = self._plugin_vars[var_name]
+                        break
+
+            # check if plugin configuration option as passed as enviornment  eg.
+            # env:
+            # - name: ANSIBLE_VALIDATE_JSONSCHEMA_DRAFT
+            if option_env_name_list and (option_name not in params):
+                for env_name_entry in to_list(option_env_name_list):
+                    if not isinstance(env_name_entry, dict):
+                        raise AnsibleError(
+                            "invalid type '%s' for the value of '%s' option,"
+                            " should to be type dict"
+                            % (type(env_name_entry), env_name_entry)
+                        )
+                    env_name = env_name_entry.get("name")
+                    if env_name in os.environ:
+                        params[option_name] = os.environ[env_name]
+                        break
+
+        argspec_result, updated_params = check_argspec(
+            yaml.dump(argspec_obj), "lookup", **params
```

Comment:
should we have the sub plugin pass it's name in, or we could see if it's set in self, so the error specifies the name of the subplugin so there's no confusion it's not in the main one?

Reply:
Good point. Fixed it